tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,186 @@
+                },
+                "adgroups": {
+                    "7086361182503780353": "2022-04-21T09:36:19.000000Z"
+                },
+                "ads": {
+                    "7086361182503780353": "2022-04-21T09:06:35.000000Z"
+                },
+                "ad_insights": {
+                    "7086361182503780353": "2021-01-25T00:00:00.000000Z"
+                }
+            }
+        }
+
+        # Set state for 2nd sync
+        menagerie.set_state(conn_id, interrupted_sync_state)
+
+        # Run sync after interruption
+        second_sync_record_count_by_stream = self.run_and_verify_sync(conn_id)
+        synced_interrupted_sync_records = runner.get_records_from_target_output()
+
+        final_state = menagerie.get_state(conn_id)
+        currently_syncing = final_state.get("currently_syncing")
+
+        # Checking that the resuming sync resulted in a successfully saved state
+        with self.subTest():
+
+            # Verify sync is not interrupted by checking currently_syncing in the state for sync
+            self.assertIsNone(currently_syncing)
+
+            # Verify bookmarks are saved
+            self.assertIsNotNone(final_state.get("bookmarks"))
+
+            # Verify final_state is equal to uninterrupted sync"s state
+            self.assertDictEqual(final_state.get("bookmarks"), first_full_sync_state.get("bookmarks"))
+
+        # Get all account ids from the config properties
+        account_ids = [x.strip() for x in self.get_credentials()["accounts"].split(",") if x.strip()]
+
+        # Stream level assertions
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                replication_key = self.expected_replication_keys()[stream]
+                # Gather actual results
+                full_records = {}
+                interrupted_records = {}
+                interrupted_record_count = second_sync_record_count_by_stream.get(stream, 0)
+                for account_id in account_ids:
+                    full_records[account_id] = [message["data"]
+                                                for message in first_full_sync.get(stream, {}).get("messages", [])
+                                                if str(message["data"]["advertiser_id"]) == account_id]
+
+                    interrupted_records[account_id] = [message["data"]
+                                                       for message in synced_interrupted_sync_records.get(stream, {}).get("messages", [])
+                                                       if str(message["data"]["advertiser_id"]) == account_id]
+
+                # Final bookmark after interrupted sync
+                # State saves stream-wise bookmarks for each account id provided in the config properties
+                for account_id in account_ids:
+                    final_stream_bookmark = final_state["bookmarks"].get(stream, {}).get(account_id)
```

Comment:
Are there no `FULL_TABLE` streams in `expected_streams`. bookmark value would be `None` for a `FULL_TABLE` and assertions on these [lines](https://github.com/singer-io/tap-tiktok-ads/blob/818cdc5e48fb1cf842a92231e252f8226177efc4/tests/test_interrupted_sync.py#L130-L131) would fail.

Reply:
For clarity, do you mean if we add new `FULL_TABLE` stream in future then this assertion would fail?

Reply:
yes

Reply:
There are no `FULL_TABLE` streams in the tap.